client/components/ReviewRating/WriteReview/WriteReview.jsx

```diff
@@ -49,132 +60,215 @@ const WriteReview = ({
-        </div>
-        <div className="input_label">
-          <span>
-            Email:
-          </span>
-          <input className="form_input" name="email" id="email" value={newReview.email || ''} onChange={onChangeHandler} required />
-        </div>
-        <div className="input_label">
-          <span>
-            Photos:
-          </span>
-          <input name="photos" id="photos" value={newReview.photos || ''} onChange={onChangeHandler} />
-          <button type="button" onClick={saveUrl}>add url</button>
-        </div>
-        {photoUrl && photoUrl.map((url) => (
-          <div key={url}>{url}</div>
-        ))}
-        <label
-          htmlFor="characteristics"
-          className="characteristics_label"
-          id="url_box"
-        >
-          {charName && charName.map((char, index) => (
-            <div key={char}>
-              <label
-                htmlFor={char}
-              >
-                {char}
-              </label>
-              <input className="form_input" value={charValue[index] || ''} name={char} id={char} type="range" min="1" max="5" onChange={onChangeHandler} />
-            </div>
+      {!isValid && (
+        <div id="error_message">
+          <div>You must enter the following:</div>
+          {errMsg[0] && errMsg[0].map((err) => (
+            <div key={err}>{err}</div>
           ))}
-        </label>
-        <div className="review_btn">
-          <button type="button" onClick={() => closeWriteReview()}>Cancel</button>
-          <button type="button" onClick={onSubmit}>Submit</button>
+          <button
+            type="button"
+            onClick={() => {
+              setErrMsg([]);
+              setIsValid(true);
+            }}
+          >
+            OK
+          </button>
         </div>
+      )}
+      <div className={isValid ? 'window' : 'close'}>
+
+        <form className="modal">
+          <h1>Write Your Review</h1>
+          <h3>
+            About the
+            {' '}
+            {productName}
+          </h3>
+          <div>
+            <span>
+              <b>Rating:</b>
+              <p />
+              <StarRating size={30} rating={selectRate} callback={setSelectRate} />
+            </span>
+          </div>
+          <div className="input_label">
+            Recommend:
+            <span>
+              Yes
+              <input name="recommend" id="yes" type="radio" value onChange={onChangeHandler} defaultChecked />
+            </span>
+            <span>
+              No
+              <input name="recommend" id="no" type="radio" value={false} onChange={onChangeHandler} />
+            </span>
+          </div>
+
+          <div className="radios">
+            {charName.map((title) => (
+              <div key={title}>
+                <span key={title}>
+                  {title}
+                </span>
+                <input name={title} type="radio" value={1 || ''} onChange={onChangeHandler} />
```

Comment:
A lot of repeated code here. Looks like a good opportunity to keep it DRY with a map.